packages/jest-resolve/src/types.ts

```diff
@@ -16,6 +16,7 @@ export type ResolverConfig = {
   modulePaths?: Array<Config.Path>;
   platforms?: Array<string>;
   resolver?: Config.Path | null;
+  asyncResolver?: Config.Path | null;
```

Comment:
ergh, that's probably a lie, since `resolver` is the replacement for `defaultResolver`... Thoughts on having it be a module that either exports a function (like today) _or_ exporting an object with `sync` and `async` properties or something? I'd like to avoid adding more configuration options, we already have so many... 😅

Reply:
I've tried it out, and also tried to explain it a bit in the docs.  Let me know what you think.

Reply:
Yeah, I think this makes more sense. I'll let it stew for a bit